common/txmgr/broadcaster.go

```diff
@@ -143,6 +143,10 @@ type Broadcaster[
 	utils.StartStopOnce
 
 	parseAddr func(string) (ADDR, error)
+
+	sequenceLock         *sync.RWMutex
```

Comment:
nit/ Mutexes are commonly embedded directly, since the zero value is usable as-is and then you don't need to initialize anything on 194.
```suggestion
	sequenceLock         sync.RWMutex
```

Reply:
Ah good to know. Just fixed it